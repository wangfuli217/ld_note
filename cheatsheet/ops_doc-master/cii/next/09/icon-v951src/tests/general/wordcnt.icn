#
#          W O R D   C O U N T I N G
#

#  This program tabulates the words in standard input and writes the
#  results with the words in a column 20 characters wide.  The definition
#  of a "word" is naive.

procedure main()
   wordcount(20)
end

procedure wordcount(n)
   local t, line, x, i
   static letters
   initial letters := &lcase ++ &ucase
   t := table(0)
   while line := read() do
      line ? while tab(upto(letters)) do
         t$<tab(many(letters))$> +:= 1
   x := sort(t,3)
   every i := 1 to *x - 1 by 2 do
      write(left(x$<i$>,n),x$<i + 1$>)
end
