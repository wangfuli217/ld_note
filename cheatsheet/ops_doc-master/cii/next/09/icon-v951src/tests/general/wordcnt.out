A                   2
B                   1
D                   1
I                   1
L                   1
N                   1
O                   2
R                   1
T                   2
This                1
U                   1
W                   1
a                   1
and                 1
characters          1
column              1
do                  3
end                 2
every               2
in                  1
initial             1
input               2
lcase               1
left                1
letters             4
line                3
local               1
main                2
many                1
n                   2
procedure           3
processes           1
results             1
scan                1
sort                1
standard            1
static              1
t                   4
tab                 2
table               1
the                 2
ucase               1
upto                1
using               1
while               1
wide                1
with                1
wordcount           2
words               1
write               1
writes              1
x                   3
y                   4
